streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -311,8 +312,10 @@ private void maybeOptimizeRepartitionOperations() {
             }
 
             final SerializedInternal serialized = new SerializedInternal(getRepartitionSerdes(entry.getValue()));
+            final String repartitionTopicName = getFirstRepartitionTopicName(entry.getValue());
```

Comment:
Now grab the topic name from the first existing repartition topic that will get merged/replaced.